core/guard/guard.go

```diff
@@ -0,0 +1,117 @@
+package guard
```

Comment:
maybe rename to `contract.go`? the files in core/guard are grouped by usage

Reply:
ok